<a href="https://colab.research.google.com/github/LarsonBodden/Project-2/blob/main/Update_Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Larson Bodden 

- Project 2
- 8/7/22

# Task 

Create a Colab notebook where you have uploaded and shown the .head() of each of your data sets.  For each of the proposed datasets, answer the following questions:

First choice: dataset 1

1) Source of data

2) Brief description of data

3) What is the target?

4) Is this a classification or regression problem?

5) How many features?

6) How many rows of data.

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

Second choice: dataset 2

1) Source of data

2) Brief description of data

3) What is the target?

4) Is this a classification or regression problem?

5) How many features?

6) How many rows of data.

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?



In [ ]:
# Mount Drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
# imports
import pandas as pd
import numpy as np
from sklearn.preprocessing import StandardScaler, OneHotEncoder
from sklearn.impute import SimpleImputer
from sklearn.compose import make_column_transformer, make_column_selector
from sklearn.pipeline import make_pipeline
from sklearn.model_selection import train_test_split
from sklearn import set_config
set_config(display='diagram')
from sklearn.linear_model import LinearRegression
from sklearn.dummy import DummyRegressor
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score
from sklearn import set_config
from sklearn.ensemble import BaggingRegressor
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import RandomForestRegressor

## Dataset 1

In [ ]:
# Load data from csv file
filename = '/content/drive/MyDrive/Coding Dojo - Data Science Bootcamp/Colab Notebooks/Projects/Project 2/Car_Purchasing_Data.csv'
df_1 = pd.read_csv(filename)
df_1.head()

Customer Name                                    Customer e-mail Country  \
0    Martina Avila  cubilia.Curae.Phasellus@quisaccumsanconvallis.edu     USA   
1    Harlan Barnes                                eu.dolor@diam.co.uk     USA   
2  Naomi Rodriquez  vulputate.mauris.sagittis@ametconsectetueradip...     USA   
3  Jade Cunningham                            malesuada@dignissim.com     USA   
4     Cedric Leach     felis.ullamcorper.viverra@egetmollislectus.net     USA   

   Gender  Age  Annual Salary  Credit Card Debt    Net Worth  \
0       0   42    62812.09301      11609.380910  238961.2505   
1       0   41    66646.89292       9572.957136  530973.9078   
2       1   43    53798.55112      11160.355060  638467.1773   
3       1   58    79370.03798      14426.164850  548599.0524   
4       1   57    59729.15130       5358.712177  560304.0671   

   Car Purchase Amount  
0          35321.45877  
1          45115.52566  
2          42925.70921  
3          67422.36313  
4          55915.46248

In [ ]:
# computing number of rows
rows = len(df_1.axes[0])
 
# computing number of columns
cols = len(df_1.axes[1])
 
print(df_1)
print("Number of Rows: ", rows)
print("Number of Columns: ", cols)

       Customer Name                                    Customer e-mail  \
0      Martina Avila  cubilia.Curae.Phasellus@quisaccumsanconvallis.edu   
1      Harlan Barnes                                eu.dolor@diam.co.uk   
2    Naomi Rodriquez  vulputate.mauris.sagittis@ametconsectetueradip...   
3    Jade Cunningham                            malesuada@dignissim.com   
4       Cedric Leach     felis.ullamcorper.viverra@egetmollislectus.net   
..               ...                                                ...   
495           Walter                                ligula@Cumsociis.ca   
496            Vanna                 Cum.sociis.natoque@Sedmolestie.edu   
497            Pearl                      penatibus.et@massanonante.com   
498             Nell                  Quisque.varius@arcuVivamussit.net   
499            Marla                          Camaron.marla@hotmail.com   

    Country  Gender  Age  Annual Salary  Credit Card Debt    Net Worth  \
0       USA       0   42 

1) Source of data
- https://www.kaggle.com/datasets/dev0914sharma/car-purchasing-model

2) Brief description of data
- This data contain the information of the various people with the following detail which can help us to find out the better prediction and also help us to train our model to predict the correct value and get a good F1 Score.

3) What is the target?
- Car Purchase Amount

4) Is this a classification or regression problem?
- This is classfication problem because i will be a predicting a discrete class label.

5) How many features?
- 6 features. I will also drop customer name and customer email because it does not add or change anything on the data set. 

6) How many rows of data.
- 500 Rows

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
- I do not forsee any challenges

## Dataset 2

In [ ]:
# Load data from Excel file
filename = '/content/drive/MyDrive/Coding Dojo - Data Science Bootcamp/Colab Notebooks/Projects/Project 2/StockX-Data-Contest-2019-3.xlsx'
df_2 = pd.read_excel(filename)
df_2.head()

Order Date   Brand                                   Sneaker Name  \
0 2017-09-01   Yeezy           Adidas-Yeezy-Boost-350-Low-V2-Beluga   
1 2017-09-01   Yeezy    Adidas-Yeezy-Boost-350-V2-Core-Black-Copper   
2 2017-09-01   Yeezy     Adidas-Yeezy-Boost-350-V2-Core-Black-Green   
3 2017-09-01   Yeezy       Adidas-Yeezy-Boost-350-V2-Core-Black-Red   
4 2017-09-01   Yeezy  Adidas-Yeezy-Boost-350-V2-Core-Black-Red-2017   

   Sale Price  Retail Price Release Date  Shoe Size  Buyer Region  
0      1097.0         220.0   2016-09-24       11.0    California  
1       685.0         220.0   2016-11-23       11.0    California  
2       690.0         220.0   2016-11-23       11.0    California  
3      1075.0         220.0   2016-11-23       11.5      Kentucky  
4       828.0         220.0   2017-02-11       11.0  Rhode Island

In [ ]:
# computing number of rows
rows = len(df_2.axes[0])
 
# computing number of columns
cols = len(df_2.axes[1])
 
print(df_2)
print("Number of Rows: ", rows)
print("Number of Columns: ", cols)

      Order Date   Brand                                   Sneaker Name  \
0     2017-09-01   Yeezy           Adidas-Yeezy-Boost-350-Low-V2-Beluga   
1     2017-09-01   Yeezy    Adidas-Yeezy-Boost-350-V2-Core-Black-Copper   
2     2017-09-01   Yeezy     Adidas-Yeezy-Boost-350-V2-Core-Black-Green   
3     2017-09-01   Yeezy       Adidas-Yeezy-Boost-350-V2-Core-Black-Red   
4     2017-09-01   Yeezy  Adidas-Yeezy-Boost-350-V2-Core-Black-Red-2017   
...          ...     ...                                            ...   
99951 2019-02-13   Yeezy    adidas-Yeezy-Boost-350-V2-Static-Reflective   
99952 2019-02-13   Yeezy    adidas-Yeezy-Boost-350-V2-Static-Reflective   
99953 2019-02-13   Yeezy    adidas-Yeezy-Boost-350-V2-Static-Reflective   
99954 2019-02-13   Yeezy    adidas-Yeezy-Boost-350-V2-Static-Reflective   
99955 2019-02-13   Yeezy    adidas-Yeezy-Boost-350-V2-Static-Reflective   

       Sale Price  Retail Price Release Date  Shoe Size  Buyer Region  
0          1097.0         2

1) Source of data
- https://www.kaggle.com/datasets/hudsonstuck/stockx-data-contest

2) Brief description of data
- The data consists of a random sample of all Off-White x Nike and Yeezy 350 sales from between 9/1/2017 (the month that Off-White first debuted “The Ten” collection) and the present. There are 99,956 total sales in the data set; 27,794 Off-White sales, and 72,162 Yeezy sales. 

3) What is the target?
- Sale Price

4) Is this a classification or regression problem?
- This is a classfication problem.

5) How many features?
- 5 Features. I will drop Order Date, and Buyer Region.

6) How many rows of data.
- 99956 Rows

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
- I do not forsee any challenges in cleaning, exploring, or modeling with this dataset